### PR TITLE
[WIP] First steps to Commands

### PR DESCRIPTION
**Pull request by joecohens**
- [x] Incidents
   - [x] ReportIncident
   - [x] ScheduleIncident
   - [x] RemoveIncident
- [x] Metrics
   - [x] AddMetric
   - [x] RemoveMetric
- [x] MetricPoints
   - [x] AddMetricPoint
   - [x] RemoveMetricPoint
- [x] Subscribers
   - [x] SubscribeSubcriber
   - [x] UnsubscribeSubcriber
   - [x] VerifySubcriber
- [x] Components
   - [x] AddComponent
   - [x] RemoveComponent
- [x] ComponentGroup
   - [x] AddComponentGroup
   - [x] RemoveComponentGroup
- [x] User
   - [x] AddTeamMember
   - [x] RemoveTeamMember
   - [x] GenerateApiToken

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Nice!

**Comment by jbrooksuk** [comment]
Do you want a list of commands?

**Comment by jbrooksuk** [comment]
We'll need a `ChangeComponentStatus` command too :)

**Comment by joecohens** [comment]
> Do you want a list of commands?

**Comment by jbrooksuk** [comment]
Rebased @joecohens

**Comment by jbrooksuk** [comment]
Couple more done @joecohens :)

**Comment by jbrooksuk** [comment]
Since we now use `RemoveComponentGroupCommandHandler`, we actually fix a bug in the component group API, where the related `group_id`'s aren't removed.

**Comment by jbrooksuk** [comment]
@joecohens!

File: app/Http/Controllers/Dashboard/IncidentController.php [hunk 1]
```diff
@@ -107,19 +111,25 @@ public function showTemplates()
...
-        $incidentData = Binput::get('incident');
-        $componentStatus = array_pull($incidentData, 'component_status');
-
-        if (array_has($incidentData, 'created_at') && $incidentData['created_at']) {
-            $incidentDate = Date::createFromFormat('d/m/Y H:i', $incidentData['created_at'], Setting::get('app_timezone'))->setTimezone(Config::get('app.timezone'));
-            $incidentData['created_at'] = $incidentDate;
-            $incidentData['updated_at'] = $incidentDate;
-        } else {
-            unset($incidentData['created_at']);
+        if ($createdAt = Binput::get('created_at')) {
```
Comment (joecohens) [inline app/Http/Controllers/Dashboard/IncidentController.php]: @jbrooksuk I would move this functionality to the handler like we did on the AddMetricPointCommandHandler, what do you think?

Comment (jbrooksuk) [inline app/Http/Controllers/Dashboard/IncidentController.php]: Hmm, I did that with the Maintenance stuff, so yeah :)

**Comment by GrahamCampbell** [comment]
Is this mergable now? We can always create issues to describe the remaining stuff to do.

**Comment by jbrooksuk** [comment]
Sure!

**Comment by joecohens** [comment]
:clap:

**Pull request merged**